gitlabform/lists/projects.py

```diff
@@ -50,27 +50,16 @@ def _get_single_project(self, target: str) -> Projects:
 
         except NotFoundException:
             debug("Could not find '%s'", target)
-            debug(
-                "Checking if it's a project that needs to be transferred from elsewhere"
-            )
-            project_transfer_source = self.configuration.config["projects_and_groups"][
+            project_transfer_source = self._get_project_transfer_source_from_config(
                 target
-            ]["project"]["transfer_from"]
+            )
 
             if project_transfer_source:
-                try:
-                    maybe_project = self.gitlab.get_project_case_insensitive(
-                        project_transfer_source
-                    )
-                    debug(
-                        "Found a project '%s' to be transferred to '%s'",
-                        maybe_project["path_with_namespace"],
-                        target,
-                    )
+                source_project = self._find_project_transfer_source_in_gitlab(
+                    project_transfer_source
+                )
+                if source_project:
                     projects.add_requested([target])
```

Comment:
I just learned about python's walrus operator (`:=`). We can re-write the nested `if` conditions like this:

```py
            if (
                project_transfer_source := self._get_project_transfer_source_from_config(
                    target
                )
            ) and self._find_project_transfer_source_in_gitlab(project_transfer_source):
                projects.add_requested([target])
```

What do you think about this? Would it make the code harder to read/understand? If it does, then we can leave it as-is.

Reply:
I think we can use that operator though I don't think it can be as simplified as that due to the error handling, updated it.

Reply:
Thanks for the update.